ehrql/query_engines/base_sql.py

```diff
@@ -411,6 +413,21 @@ def get_sql_to_first_of_month(self, node):
     def to_first_of_month(self, date):
         raise NotImplementedError()
 
+    @get_sql.register(Function.MaximumOf)
+    def get_sql_maximum_of(self, node):
+        return self.get_nary_function(node, greatest)
+
+    @get_sql.register(Function.MinimumOf)
+    def get_sql_minimum_of(self, node):
+        return self.get_nary_function(node, least)
+
+    def get_nary_function(self, node, function):
+        sources = [*node.sources]
+        columns = [self.get_expr(s) for s in sources]
+        types = {c.type for c in columns}
+        function.type = types.pop()
```

Comment:
Does passing the type as keyword argument work? e.g.
```py
return function(*columns, type=types.pop())
```
Mutating the shared object feels a bit icky, but probably won't cause any actual problems.

Reply:
it doesn't, alas. This was the ickiness I was referring to on Slack.